### ISSUE TITLE
E-Mail Validation is wrong

### ISSUE THREAD

**Issue by juergrinaldi** [zitadel/zitadel#3919]

**Comment by mffap** [comment]
Same here. I encountered the same thing when doing testing
![image](https://user-images.githubusercontent.com/1366906/187686858-870140f2-59b5-4a53-9294-f9cbb2ac95fa.png)

**Comment by mffap** [comment]
And another thing: 

The dialog "Change Email" on a user is EndUser facing and the email address should be validated in the GUI in my opinion, which is currently not done. I can enter a wrong email and get only a notification after clicking save, that the email might be wrong. This should be validated before being able to send.

![image](https://user-images.githubusercontent.com/1366906/187687294-f949a9cb-1c2f-4b10-af98-09ab53886a76.png)

**Comment by livio-a** [comment]
`est@test`, `mpa+ux20220831`, ... are technically valid email addresses according to specification. It defines the following schema:
`<local part>@<domain part>` where the domain part mainly follows DNS specification, but also IPs are allowed (in `[]`).
See https://www.rfc-editor.org/rfc/rfc5321#section-4.1.3

IMO the question is if we need to support those cases, e.g. for on-prem instances)

**Comment by mffap** [comment]
> `est@test`, `mpa+ux20220831`, ... are technically valid email addresses according to specification. It defines the following schema: `<local part>@<domain part>` where the domain part mainly follows DNS specification, but also IPs are allowed (in `[]`). See https://www.rfc-editor.org/rfc/rfc5321#section-4.1.3
> 
> IMO the question is if we need to support those cases, e.g. for on-prem instances)

Agreed for self-hosted. On zitadel.cloud this is a Bug in my view and should not be possible.

**Comment by altissia-be** [comment]
Any update on this? I was unable to find a list of email validation rules to implement on my side to match Zitadel's behavior.  
Emails that should be ok are rejected, and I need to make sure to reject them before accepting the user input, the same way the frontend here also lacks validation. Or Zitadel should allow these emails.

Ideally we should be able to configure it: best would be to have validation that follows the RFCs and allows the given examples on [Wikipedia](https://en.wikipedia.org/wiki/Email_address#Valid_email_addresses). But a more permissive/minimalist option of simply checking the existence of an @ would also cover our use case.  
Users emails are validated through an email link anyway, so not validating it matches the actual rules is not an issue.
